waiter/config-full.edn

```diff
@@ -1001,6 +1001,14 @@
  ;; Components also get passed in the :kv-store-factory, :leader?-fn and :synchronize-fn for use in the factory.
  :custom-components {:in-memory-kv-store {:factory-fn waiter.kv/new-local-kv-store}}
 
+ ;; Configure event maintainer that handles internal events to be sent externally
+ :event-maintainer-config {
+                           ;; Buffer size of the event-chan where other threads can push internal events to
+                           :event-chan-buffer-size 1000
+
+                           ;; The fn that handles new instance-failure events
```

Comment:
We should document the signature. I'm also wondering if this should be a component with an interface (i.e. defprotocol) so it can be initialized with some nested configuration, e.g. how we use `kind` and initialize other components. E.g. a components that will make http-calls may need additional configuration.